pkg/ceph/osd/device.go

```diff
@@ -305,6 +288,71 @@ func partitionOSD(context *clusterd.Context, config *osdConfig) error {
 	return nil
 }
 
+func prepareFilestoreDevice(context *clusterd.Context, config *osdConfig, doFormat bool) error {
+	if !isFilestoreDevice(config) {
+		// nothing to do
+		return nil
```

Comment:
shouldnt this raise an error if it gets here and it is not Filestore? Or are there valid cases where this could happen?